src/go/rpk/pkg/config/weak.go

```diff
@@ -533,6 +536,64 @@ func (s *ServerTLS) UnmarshalYAML(n *yaml.Node) error {
 	return nil
 }
 
+func (ss *SeedServer) UnmarshalYAML(n *yaml.Node) error {
+	var internal struct {
+		// New schema should only contain Address and Port, but we will
+		// support this under Host also.
+		Address weakString     `yaml:"address"`
+		Port    weakInt        `yaml:"port"`
+		Host    *SocketAddress `yaml:"host"`
+		// deprecated
+		NodeID *weakInt `yaml:"node_id"`
+	}
+	if err := n.Decode(&internal); err != nil {
+		return err
+	}
+	if internal.NodeID != nil {
+		fmt.Println("redpanda yaml: redpanda.seed_server.node_id is deprecated and unused")
+	}
+
+	if internal.Address != "" || internal.Port != 0 {
```

Comment:
I thought this block might like something like this, where `Host` is not an inner pointer:

```go
embedded := SocketAddress{string(internal.Address), int(internal.Port)}
nested := internal.Host

embeddedZero := reflect.DeepEqual(embedded, SocketAddress{})
nestedZero := reflect.DeepEqual(nested, SocketAddress{})

if !embeddedZero && !nestedZero && !reflect.DeepEqual(embedded, nested) {
  return errors.New("redpanda.yaml redpanda.seed_server: nested host differs from address and port fields; only one must be set")
}

ss.Host = embedded
if embeddedZero {
  ss.Host = nested
}
return nil
```

Basically, if both are present, they must be equal. If either are zero, then it doesn't matter. It simplifies the logic and the checks we need to make. What do you think?

Reply:
I think I got the requirement completely wrong haha. Changed and changed the unit tests.